### PR TITLE
use zap logger

### PR DESCRIPTION
**Pull request by SvenDowideit**
for #139

### CODE REVIEW

**Comment by lucaslorentz** [comment]
@SvenDowideit  That's great!

I fixed tests after Zap Logger changes:
https://github.com/SvenDowideit/caddy-docker-proxy/pull/1

But there are still some things I'm concerned about, I will review further the PR after you incorporate my changes

**Comment by lucaslorentz** [comment]
My main concern now is that before Zap all logs from the generator were going to a buffer, and the logs were printed only if they are different from the last generator run.

That was avoiding printing the same logs every 30 seconds.

We can probably implement something similar with Zap, by creating temporary sinks or something like that.

But maybe we should just drop that feature and log everything as it happens, probably most applications do that.

@SvenDowideit @francislavoie What's your view on this?

**Comment by francislavoie** [comment]
I think it's useful to always log when events happen. If it's too noisy, the user can up the log level, or `exclude` some of the logs being emitted by module name.

File: plugin/cmd.go [hunk 1]
```diff
@@ -62,7 +63,7 @@ func cmdFunc(flags caddycmd.Flags) (int, error) {
 	options := createOptions(flags)
 
 	if options.Mode&config.Server == config.Server {
-		log.Printf("[INFO] Running caddy proxy server")
+		logger().Info("Running caddy proxy server")
```
Comment (lucaslorentz) [inline plugin/cmd.go:66]: Should we cache this logger in global constant instead of recreating it every time we need it?

Comment (lucaslorentz) [reply plugin/cmd.go:66]: Maybe not globally, but at least locally
Because caddy will change the logger every time a caddyfile is reloaded.

Comment (SvenDowideit) [reply plugin/cmd.go:66]: yup, one mo :)

**Comment by SvenDowideit** [comment]
so from what I'm seeing in my test system, if nothing changes, and you havn't set the caddy `debug`, there is no log output unless a change happens. when you enable debug, you get lots of info

(this was actually the reason I went down this for my other PR - I wanted to be able to see everything all the time if I was debugging a weird template issue)

**Comment by lucaslorentz** [comment]
Just did 2 tests.

One is adding a container with `label: caddy.log.format=console` and then removing it.
Log changed nicely between JSON, Console and then back to JSON. ✅ 

And another is to start caddy container already with label `label: caddy.log.format=console`. Logs starts as JSON and after some time switches to console. ✅ 

The only thing I'm missing now is a way to configure logging from the beginning as console for example. But we can do that later.

**Comment by SvenDowideit** [comment]
oh! awesome, thanks, after this, my week got busy - I didn't even know about the log format, will have to go look further :)

**Pull request merged**